docs/dev/local-transaction-design.md

```diff
@@ -0,0 +1,19 @@
+# Local Transaction Design
+
+The broker core handles both transactional and non-transactional in a unified way implementing BrokerTransaction 
+interface. AutoCommitTransaction is the non-transactional implementation and LocalTransaction is the local transactional 
+implementation. AMQP transport register tx.select, tx.commit, tx.rollback and broker core act based on the relevant 
+operation. 
+
+Message acknowledge considering as dequeue and message publish considering as enqueue. AutoCommitTransaction directly 
+invokes the Broker API on dequeue or enqueue operation and there is no effect on commit or rollback. If somebody calls 
+tx.commit or tx.rollback on the non-transactional channel, it'll end up in channel exception as per the spec.
```

Comment:
Shall we use the written form (i.e. "It will" instead of "it'll")

Reply:
Updated.